### PR TITLE
[pileup_encoder] indel encoding for pileups

### LINKED ISSUE #77: [pileup_encoder] add encoding for indels
encode indels into pileup tensor following same principle as for SNPs.

i.e. keeping the variant location of interest at the center of the pileup

### PR DESCRIPTION
**Pull request by tijyojwad**
Adds encoding for indels, and support for naive visualization of encodings to debug.

Updated necessary tests.

Closes #77

### CODE REVIEW

File: variantworks/sample_encoder.py [hunk 1]
```diff
@@ -153,7 +158,7 @@ def _fill_layer(self, layer, pileupread, left_offset, right_offset, row, pileup_
                     qual = qual / MAX_BASE_QUALITY
                 tensor[row, pileup_pos] = qual
         elif layer == self.Layer.MAPPING_QUALITY:
-            MAX_MAPPING_QUALITY = 50.0
+            MAX_MAPPING_QUALITY = 100.0
```
[the remark this answers is not in the document]
Comment (tijyojwad) [reply variantworks/sample_encoder.py:161]: It's a good question @ohadmo . In general it's good to normalize values when feeding to DL models in order to keep the input value range similar. e.g. if MAPQ values can go up to 100, and other values only go up to 10, then the scale of the values can skew the learning and implicitly make those features have more weight. But I think the right API is to provide a flag to normalize or not, and we'll add that over time. Does that help?

[the remark this answers is not in the document]
Comment (ohadmo) [reply variantworks/sample_encoder.py:161]: @tijyojwad  Let me rephrase, According to the SAM Format specification, theoretically, the MAPQ values can be positive infinite. What I'm asking is, in practice, what is the distribution of the MAPQ values? since `MAPQ=100  --> Pr{mapping position is wrong} ~ 10xe^11` which seems to me unlikely...

[the remark this answers is not in the document]
Comment (tijyojwad) [reply variantworks/sample_encoder.py:161]: Now I understand - `MAPQ = 100 => Pr{error} = 1e-10`. Which is indeed very small. When I read up online it seemed to me that tools limit MAPQ values to some number (42 seemed like a common one by BWA I think). So I left it at 50. But then I came across a dataset (not sure how it's generated), that had a MAPQ of 70 something. `100` seemed to be an appropriate upper bound (though you're right that theoretically it could be infinite when `Pr{error} = 0`). I intend to make this a configurable parameter soon.

[the remark this answers is not in the document]
Comment (tijyojwad) [reply variantworks/sample_encoder.py:161]: Created task https://github.com/clara-parabricks/VariantWorks/issues/85 to track configurability of layers
